js/fetch.ts

```diff
@@ -399,6 +401,28 @@ export async function fetch(
     headers = input.headers;
   }
 
+  // Fetching a local file
+  // See RFC8089 : https://tools.ietf.org/html/rfc8089
+  const m = url.match(/file:\/{1,5}(.*)/i);
+  if (m && m[0]) {
+    try {
+      const f = await open(m[1]);
+      const headers: [string, string][] = [];
+      // TODO : Add return type. Using Deno_std media_types module
+      const response = new Response(200, headers, f.rid);
+      return response;
+    } catch (e) {
+      if (
+        e instanceof DenoError &&
+        (e as DenoError<ErrorKind.NotFound>).kind === ErrorKind.NotFound
+      ) {
+        return new Response(404, [], 0);
+      } else {
+        return new Response(500, [], 0);
```

Comment:
Is there any metadata associated with this error? If so, would it make sense to JSON-encode it and put it as the body of this response?

Reply:
As far as i know the errorkind notfound is just an enum for the error code. The only error i think would be not found or permission error. But just put a standard json encoded object like `{message:"file not found"}` would be interesting

Reply:
Would calling `JSON.stringify(e)` be a possibility (with try catch)?